File: app/src/main/java/com/teixeira0x/subtypo/utils/ToastUtils.java
package com.teixeira0x.subtypo.utils;

import android.widget.Toast;
import com.teixeira0x.subtypo.App;

public class ToastUtils {

  public static void showShort(int message, Object... args) {
    showShort(App.getInstance().getString(message, args));
  }

  public static void showShort(String message) {
    Toast.makeText(App.getInstance(), message, Toast.LENGTH_SHORT).show();
  }

  public static void showLong(int message, Object... args) {
    showLong(App.getInstance().getString(message, args));
  }

  public static void showLong(String message) {
    Toast.makeText(App.getInstance(), message, Toast.LENGTH_LONG).show();
  }
}

File: app/src/main/java/com/teixeira0x/subtypo/ui/utils/ToastUtils.kt
package com.teixeira0x.subtypo.ui.utils

import android.content.Context
import android.widget.Toast

fun Context.showToastLong(message: Int) {
  showToast(getString(message), Toast.LENGTH_LONG)
}

fun Context.showToastLong(message: String) {
  showToast(message, Toast.LENGTH_LONG)
}

fun Context.showToastShort(message: Int) {
  showToast(getString(message), Toast.LENGTH_SHORT)
}

fun Context.showToastShort(message: String) {
  showToast(message, Toast.LENGTH_SHORT)
}

fun Context.showToast(message: String, duration: Int) =
  Toast.makeText(this, message, duration).show()
